server-common/src/main/java/org/apache/gravitino/server/authorization/jcasbin/JcasbinAuthorizer.java

```diff
@@ -0,0 +1,159 @@
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.Principal;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.commons.io.IOUtils;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.auth.AuthConstants;
+import org.apache.gravitino.authorization.Privilege;
+import org.apache.gravitino.server.authorization.GravitinoAuthorizer;
+import org.apache.gravitino.server.authorization.MetadataIdConverter;
+import org.apache.gravitino.server.web.ObjectMapperProvider;
+import org.apache.gravitino.storage.relational.po.RolePO;
+import org.apache.gravitino.storage.relational.po.SecurableObjectPO;
+import org.apache.gravitino.storage.relational.service.MetalakeMetaService;
+import org.apache.gravitino.storage.relational.service.RoleMetaService;
+import org.apache.gravitino.storage.relational.service.UserMetaService;
+import org.casbin.jcasbin.main.Enforcer;
+import org.casbin.jcasbin.model.Model;
+
+/** The Jcasbin implementation of GravitinoAuthorizer. */
+public class JcasbinAuthorizer implements GravitinoAuthorizer {
+
+  /** Jcasbin enforcer is used for metadata authorization. */
+  private Enforcer enforcer;
+
+  /**
+   * loadedRoles is used to cache roles that have loaded permissions. When the permissions of a role
+   * are updated, they should be removed from it.
+   */
+  private Set<Long> loadedRoles = ConcurrentHashMap.newKeySet();
+
+  @Override
+  public void initialize() {
+    try (InputStream modelStream =
+        JcasbinAuthorizer.class.getResourceAsStream("/jcasbin_model.conf")) {
+      Preconditions.checkNotNull(modelStream, "Jcasbin model file can not found.");
+      String modelData = IOUtils.toString(modelStream, StandardCharsets.UTF_8);
+      Model model = new Model();
+      model.loadModelFromText(modelData);
+      enforcer = new Enforcer(model, new PassThroughAdapter());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public boolean authorize(
+      Principal principal,
+      String metalake,
+      MetadataObject metadataObject,
+      Privilege.Name privilege) {
+    return authorizeInternal(principal, metalake, metadataObject, privilege.name());
+  }
+
+  @Override
+  public boolean isOwner(Principal principal, String metalake, MetadataObject metadataObject) {
+    return authorizeInternal(principal, metalake, metadataObject, AuthConstants.OWNER);
+  }
+
+  @Override
+  public void handleRolePrivilegeChange(Long roleId) {
+    loadedRoles.remove(roleId);
+    enforcer.deleteRole(String.valueOf(roleId));
+  }
+
+  @Override
+  public void close() throws IOException {}
+
+  private boolean authorizeInternal(
+      Principal principal, String metalake, MetadataObject metadataObject, String privilege) {
+    String username = principal.getName();
+    Long metalakeId = MetalakeMetaService.getInstance().getMetalakeIdByName(metalake);
```

Comment:
It's not good to use `MetalakeMetaServie` instead EntityStore interface. Because we want to use EntityStore interface to provide universial access .